### PR TITLE
Aptos accumulators

### PR DESCRIPTION
**Pull request by m30m**

### CODE REVIEW

File: target_chains/aptos/contracts/sources/merkle.move
Comment (optke3) [inline target_chains/aptos/contracts/sources/merkle.move:39]: should one check that the references are equal, or that the underlying bytes are equal?

File: target_chains/aptos/contracts/sources/keccak160.move
Comment (jayantk) [inline target_chains/aptos/contracts/sources/keccak160.move:61]: maybe test some cases where it should return false

File: target_chains/aptos/contracts/sources/merkle.move
Comment (jayantk) [inline target_chains/aptos/contracts/sources/merkle.move:30]: root could also be a Hash right?

File: target_chains/aptos/contracts/sources/merkle.move
Comment (jayantk) [inline target_chains/aptos/contracts/sources/merkle.move:57]: add some tests for `check` also

File: target_chains/aptos/contracts/sources/pyth.move
Comment (jayantk) [inline target_chains/aptos/contracts/sources/pyth.move:183]: please document these methods. it's not very clear what they do (especially since the signatures are all vec<u8>)

File: target_chains/aptos/contracts/sources/pyth.move
Comment (jayantk) [inline target_chains/aptos/contracts/sources/pyth.move:248]: I think this check shouldn't be here -- I believe a bump of the minor version corresponds to a forward-compatible message change (like extending the length of a part of the payload) where this code should be able to deserialize it.

File: target_chains/aptos/contracts/sources/pyth.move
Comment (jayantk) [inline target_chains/aptos/contracts/sources/pyth.move:486]: leave a TODO that this may be expensive and need to be optimized. (this is a general annoying thing about programming on the blockchain, which is that you have to be very sensitive to computation cost. which often forces you to implement your code in a less obvious way)

File: target_chains/aptos/contracts/sources/pyth.move
Comment (jayantk) [inline target_chains/aptos/contracts/sources/pyth.move:974]: I suggest adding a forward-compatibility check too where we generate a message payload that updates the minor version + extends all of the extensible parts of the message. (this can be done in a subsequent PR though)

File: .github/workflows/aptos-contract.yml
Comment (jayantk) [inline .github/workflows/aptos-contract.yml:30]: thank you 🙇

**Review by jayantk — APPROVED** [review]
LGTM! I think there is one minor bug here, but should be easy to fix.

Also, as a general FYI, if I approve a PR with comments, it means "please address the comments (they are easy) before merging, but no need to request another review from me"

File: target_chains/aptos/contracts/sources/pyth.move
Comment (ali-bahjati) [inline target_chains/aptos/contracts/sources/pyth.move:222]: i think it's good to prefix some of the methods with parse_and_verify_... for more clarity.

**Review by ali-bahjati — APPROVED** [review]
Nice!

**Pull request merged**
